### DISCUSSION TITLE
HMR takes 4 seconds in a large Nuxt app

### DISCUSSION THREAD

**Discussion by martinszeltins** [nuxt/nuxt#26442] [Questions]
I have a large Nuxt 3 app with lots of packages and modules installed and HMR is taking 4 seconds which is a super long time. I don't have a clue how to even begin debugging the cause for this or how to make it fast again. Any ideas where to start?

With a fresh installation it is super fast but the more packages and code I add to my app the slower it gets.

**Comment by kstraszewski** [comment]
I've manage large fullstack app, and was working with some medium-sized ones, and have the same issue, apps takes sometimes 12sec to HMR, would love to know how can we fix it, coz poor development performance is my only issue with nuxt, but it starts to be very problematic to maintain
